### PR TITLE
Import Models from URL

### PR DESCRIPTION
**Pull request by dselman**
## Checklist
 - [x]  A link to the issue/user story that the pull request relates to
 - [x]  How to recreate the problem without the fix
 - [x]  Design of the fix
 - [x]  How to prove that the fix works
 - [x]  Automated tests that prove the fix keeps on working
 - [x]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story
#579 

## Steps to Reproduce
1. See the issue

## Existing issues

- [x] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [x] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [x] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Design of the fix
See issue

## Validation of the fix
Manual verification that an archive can be created with external models, that it can be serialized to an archived, persisted and reopened and that the external models are preserved.

## Automated Tests
100% unit test coverage.

## What documentation has been provided for this pull request
See #579 for skeletal text for inclusion in the documentation.

### CODE REVIEW

**Comment by dselman** [comment]
TODO:

- [ ] Issue with browserifying request module? Missing module `net`?
- [ ] Broken doc links for ModelFile
- [ ] Make updateExternalModels an option for `fromDirectory`
- [ ] Add updateExternalModels flag to `composer archive create` and pass additional options
- [ ] Write skeltal docs, coordinate with @EdProsser

**Comment by dselman** [comment]
Added documentation to issue #579

**Comment by dselman** [comment]
@sstone1 can you override the DCO please -- I think I squashed too many commits here.

File: packages/composer-common/lib/introspect/loaders/githubmodelfileloader.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Load a ModelFile from a URL and return it
+     * @param {string} url - the url to get
+     * @param {object} options - additional options
+     * @return {Promise} a promise to the ModelFile
+     */
+    load(url, options) {
+        const rewrittenUrl = 'https://raw.githubusercontent.com/' + url.substring(9);
```
Comment (sstone1) [inline packages/composer-common/lib/introspect/loaders/githubmodelfileloader.js]: I think we'll probably end up having to extend this URL scheme and this loader in the future to cope with GitHub Enterprise instances but it's fine as-is.

File: packages/composer-common/lib/introspect/loaders/httpmodelfileloader.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const axios = require('axios');
```
Comment (sstone1) [inline packages/composer-common/lib/introspect/loaders/httpmodelfileloader.js]: Do we really need yet another HTTP request module? We already use `request` and `xhr`.

File: packages/composer-common/lib/introspect/loaders/jobqueue.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    /* eslint-disable class-methods-use-this */
+
+    /* istanbul ignore next */
+    /**
+     * Abstract method: must be overriden.
+     * @param {any} job - data for the job
+     * @return {Promise} promise to the job results.
+     */
+    runJob(job) { return Promise.resolve(job); }
```
Comment (sstone1) [inline packages/composer-common/lib/introspect/loaders/jobqueue.js]: I think if you add the `@abstract` tag to the JSDoc comment you don't need any of these eslint/istanbul hacks 😉

File: packages/composer-website/jekylldocs/tutorials/deploy-to-fabric-multi-org.md [hunk 4]
```diff
@@ -313,6 +313,12 @@ copy the contents of the file `/tmp/composer/ca-orderer.txt` and replace the tex
 
 Once done, save this file as `/tmp/composer/byfn-network.json`.
 
+Run the following commands to create connection profiles for Org1 and Org2:
```
Comment (sstone1) [inline packages/composer-website/jekylldocs/tutorials/deploy-to-fabric-multi-org.md]: What's this doing in this PR?

**Pull request closed without merging**